lib/segment/src/vector_storage/mmap_vectors.rs

```diff
@@ -34,7 +34,9 @@ fn open_read(path: &Path) -> OperationResult<Mmap> {
         .create(true)
         .open(path)?;
 
-    Ok(unsafe { MmapOptions::new().map(&file)? })
+    let mmap = unsafe { MmapOptions::new().map(&file)? };
```

Comment:
`Advice` option is enabled only on Unix systems. It's required to add `#[cfg(unix)]` and don't advise on other systems

Reply:
@generall, @IvanPleshkov 

Regarding the `#[cfg(unix)]`, what if:

- we define our own `pub enum Advice` with the same variants as `memmap2::Advice`, _but without `#[cfg(unix)]`_
  - we can also implement `str::FromStr`/`fmt::Display` and/or `serde::Deserialize`/`serde::Serialize` for our `Advice`
- and `pub trait Madviceable` (with a `fn madvice(&self, advice: Advice)` method) to generalize over `memmap2::Mmap` and `memmap2::MmapMut`
  - and, maybe, a `pub fn madvice(mmap: &impl Madviceable, advice: Advice)`?
- and on `#[cfg(not(unix))]` platforms `Madviceable::madvice` will simply be a no-op
- this way we don't have to sprinkle `#[cfg(unix)]` all over the code

Reply:
For the life of me, I just can't spell A-D-V-I-S-E properly. 🙈

Reply:
I like your idea, @generall. Your opinion is good for high-level design and I like it. Just notice, that `memmap2` crate covers by `#[cfg(unix)]` all methods related to `Advice`

Reply:
Done.